hazelcast/src/main/java/com/hazelcast/client/impl/protocol/task/NoSuchMessageTask.java

```diff
@@ -44,8 +44,10 @@ protected ClientMessage encodeResponse(Object response) {
 
     @Override
     protected void processMessage() {
-        logger.warning("Unrecognized client message received with type: 0x"
-                + Integer.toHexString(parameters.getMessageType()));
+        String message = "Unrecognized client message received with type: 0x"
+                + Integer.toHexString(parameters.getMessageType());
+        logger.warning(message);
+        throw new UnsupportedOperationException(message);
```

Comment:
Is caller going to receive `UnsupportedOperationException` directly? Or will it be wrapped inside another exception?